packages/beacon-node/src/network/reqresp/beaconBlocksMaybeBlobsByRoot.ts

```diff
@@ -46,16 +54,26 @@ export async function unavailableBeaconBlobsByRoot(
   config: ChainForkConfig,
   network: INetwork,
   peerId: PeerIdStr,
-  unavailableBlockInput: BlockInput,
+  unavailableBlockInput: BlockInput | NullBlockInput,
   metrics: Metrics | null
 ): Promise<BlockInput> {
-  if (unavailableBlockInput.type !== BlockInputType.blobsPromise) {
-    return unavailableBlockInput;
+  if (unavailableBlockInput.block !== null && unavailableBlockInput.type !== BlockInputType.blobsPromise) {
+    return unavailableBlockInput as BlockInput;
   }
 
-  const blobIdentifiers: deneb.BlobIdentifier[] = [];
-  const {block, blobsCache, resolveAvailability, blockBytes} = unavailableBlockInput;
+  // resolve the block if thats unavailable
+  let block, blobsCache, blockBytes, resolveAvailability;
+  if (unavailableBlockInput.block === null) {
+    const allBlocks = await network.sendBeaconBlocksByRoot(peerId, [fromHexString(unavailableBlockInput.blockRootHex)]);
```

Comment:
now the block source `lodestar_import_block_by_source_total` is very tricky, it could be wrong for now

for block, we want to know if it comes from gossip, or this function, or the above `beaconBlocksMaybeBlobsByRoot()` function

same for blobs, we want to know % of them coming from 3 different places as above

if it's too much for this PR please have a follow up PR or track in another issue @g11tech